sdk/python/lib/pulumi/provider/provider.py

```diff
@@ -108,3 +124,12 @@ def call(self, token: str, args: Inputs) -> CallResult:
         """
 
         raise Exception(f"Unknown method {token}")
+
+    def invoke(self, token: str, args: Inputs) -> InvokeResult:
```

Comment:
Also wondering if `args` should be typed as `Mapping[str, Any]` since the inputs for an invoke aren't going to be `Input[T]`.

We do make the distinction in the TS provider:

https://github.com/pulumi/pulumi/blob/7c01f1ba05c133002dd448e0d61c99fff8c52210/sdk/nodejs/provider/provider.ts#L213-L227